application/helpers/frontend_helper.php

```diff
@@ -2214,3 +2214,29 @@ function cookieConsentLocalization()
         gT('Please be patient until you are forwarded to the final URL.')
     );
 }
+
+/**
+ * Returns an array of URL parameters that can be forwarded
+ *
+ * @return array<string,mixed>
+ */
+function getForwardParameters()
+{
+    $reservedGetValues = array(
+        'token',
+        'sid',
+        'gid',
+        'qid',
+        'lang',
+        'newtest',
+        'action',
+        'seed',
+        'index.php',
+    );
+
+    $parameters = [];
+    if (in_array(Yii::app()->getRequest()->getRequestType(), ['GET', 'POST'])) {
```

Comment:
Gonna be a bit anal here and ask you to supply `$request` as input argument to the function. This will make it testable. Also replace $_GET below with $request->getQueryParams()... Or not! Since Yii 1 does not have that function. D:

Reply:
https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L659

Reply:
Not reading globals is part of this guideline: https://manual.limesurvey.org/Code_quality_guide#Push_side-effects_up_in_the_stack_trace

Reply:
> https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L659

yii**2** none in Yii 1, must be created.

Reply:
Why, @Shnoulle?

Reply:
Maybe we should ask how it looks in Yii 3 instead :d

Reply:
> Why, @Shnoulle?

? I don't understand the question. We use Yii**1**, if need a function not in Yii1 : we need to create it.

Reply:
> We use Yii**1**, if need a function not in Yii1 : we need to create it.

Yes. And in this case, it already existed in Yii 2. :) So we can just copy it?

Reply:
Oh wait, I misunderstood. I think.

Reply:
I know I linked to Yii 2.

> Since Yii 1 does not have that function

Reply:
getQueryParams is part of PSR, so it's future proof for Yii 3. @gabrieljenik Can you add that method to the LS request object, and the refactor this function to use it instead of accessing $_GET?

Reply:
Great idea to copy/paste some code from Yii3